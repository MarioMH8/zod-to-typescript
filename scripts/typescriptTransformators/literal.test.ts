import { ZodTypescriptTransformator } from "@scripts/index";
import { z as zod } from "zod";

describe("literal", () => {
	it("string", () => {
		const result = ZodTypescriptTransformator.convert(zod.literal("duplojs"));

		expect(result).toMatchSnapshot();
	});

	it("number", () => {
		const result = ZodTypescriptTransformator.convert(zod.literal(42));

		expect(result).toMatchSnapshot();
	});

	it("boolean", () => {
		const result = ZodTypescriptTransformator.convert(zod.literal(true));

		expect(result).toMatchSnapshot();

		const result2 = ZodTypescriptTransformator.convert(zod.literal(false));

		expect(result2).toMatchSnapshot();
	});

	it("null", () => {
		const result = ZodTypescriptTransformator.convert(zod.literal(null));

		expect(result).toMatchSnapshot();
	});

	it("undefined", () => {
		const result = ZodTypescriptTransformator.convert(zod.literal(undefined));

		expect(result).toMatchSnapshot();
	});

	it("bigint", () => {
		const result = ZodTypescriptTransformator.convert(zod.literal(42n));

		expect(result).toMatchSnapshot();
	});

	it("symbol", () => {
		const result = ZodTypescriptTransformator.convert(zod.literal(Symbol("symbol")));

		expect(result).toMatchSnapshot();
	});

	it("force error", () => {
		const func = () => "duplojs";
		const result = ZodTypescriptTransformator.convert(zod.literal(func as never));

		expect(result).toMatchSnapshot();
	});
});
