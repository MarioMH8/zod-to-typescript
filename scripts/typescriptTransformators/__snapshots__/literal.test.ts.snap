// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`literal > bigint 1`] = `"type Zod2ts_6_duplojs = 42;"`;

exports[`literal > boolean 1`] = `"type Zod2ts_2_duplojs = true;"`;

exports[`literal > boolean 2`] = `"type Zod2ts_3_duplojs = false;"`;

exports[`literal > force error 1`] = `"type Zod2ts_8_duplojs = unknown;"`;

exports[`literal > null 1`] = `"type Zod2ts_4_duplojs = null;"`;

exports[`literal > number 1`] = `"type Zod2ts_1_duplojs = 42;"`;

exports[`literal > string 1`] = `"type Zod2ts_0_duplojs = "duplojs";"`;

exports[`literal > symbol 1`] = `"type Zod2ts_7_duplojs = "Symbol(symbol)";"`;

exports[`literal > undefined 1`] = `"type Zod2ts_5_duplojs = undefined;"`;
